src/Symfony/Component/RateLimiter/Tests/Util/TimeUtilTest.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Symfony\Component\RateLimiter\Tests\Util;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\RateLimiter\Util\TimeUtil;
+
+class TimeUtilTest extends TestCase
+{
+    public function provideTimeUtilData(): \Generator
+    {
+        yield ['PT5S', 5];
+
+        yield ['PT1M5S', 65];
+
+        yield ['PT2H4M', 2 * 3600 + 4 * 60];
+
+        yield ['P1Y', 24 * 3600 * 365];
```

Comment:
I would expect that this test will start to fail on March 1st, 2023. 🤓

Reply:
I'm struggling a bit with these tests, as months and years, of course, may have different number of days. Any advice welcome about this one!

Reply:
Is the variable length of the limit desired? So far, the rate limiter worked with fixed lengths, IIRC. We would change that now.

Reply:
Also, and given other tests, I'm wondering if it's worth testing this method doing a simple timestamp operation. What I mean is if it changes, FixedWindowLimiter tests and similar will break, as I updated them with long intervals. The lack of tests on this method currently makes me wonder even more.